1120: Suppress websocket aborted error log

baemyung opened the pull request:
When websocket is aborted under some situations like GUI console is aborted earlier before websocket write is still waiting for completion, it is currently logged as ERROR. However, it may not need to be an error. This commit will change it as WARNGING.

For example,
```
Oct 31 10:40:29 balco10 bmcwebd[1230]: [websocket_impl.hpp:305] Error in ws.async_write Operation canceled [system:125 at /usr/include/boost/beast/websocket/impl/stream_impl.hpp:355:13 in function 'bool boost::beast::websocket::stream< <template-parameter-1-1>, <anonymous> >::impl_type::check_stop_now(boost::beast::error_code&)'] 
```

Tested:
-  While GUI page is loading pages (e.g. pcie topology), close web-browser multiple times.

Change-Id: I4b16c21d2784ec0b9774ab256f833bb38fc34a27

baemyung committed:
Suppress websocket aborted error log

When websocket is aborted under some situations like GUI console is
aborted earlier before websocket write is still waiting for completion,
it is currently logged as ERROR. However, it may not need to be an
error. This commit will change it as WARNING.

For example,

```
Oct 31 10:40:29 balco10 bmcwebd[1230]: [websocket_impl.hpp:305] Error in ws.async_write Operation canceled [system:125 at /usr/include/boost/beast/websocket/impl/stream_impl.hpp:355:13 in function 'bool boost::beast::websocket::stream< <template-parameter-1-1>, <anonymous> >::impl_type::check_stop_now(boost::beast::error_code&)']
```

Tested:
-  While GUI page is loading pages (e.g. pcie topology), close
   web-browser and check the bmcweb journal records.

Change-Id: I4b16c21d2784ec0b9774ab256f833bb38fc34a27
Signed-off-by: Myung Bae <myungbae@us.ibm.com>

baemyung commented:
Upstream: https://gerrit.openbmc.org/c/openbmc/bmcweb/+/84819

abiasojo reviewed (approved):
It looks fine to me

baemyung committed:
Break out lambdas in websocket

Move out the large lambdas into normal methods to maintain more easily.

Tested:
- Unit tests pass

Change-Id: I9450af36d45b2b17e8a063f383d91026db581d27
Signed-off-by: Myung Bae <myungbae@us.ibm.com>

jeaaustx commented on http/websocket_impl.hpp:
```diff
@@ -256,6 +256,10 @@ class ConnectionImpl : public Connection
             {
                 BMCWEB_LOG_ERROR("doRead error {}", ec);
             }
+            else if (ec == boost::asio::error::operation_aborted)
```
Doesn't this need to be above the L253-L255 check? (I.e. that check is for != which will fire for this. Seems like we would end up with both an ERROR and a WARNING message for the operation_aborted case.)

baemyung replied:
Thanks for finding this.
The fix is submitted via https://gerrit.openbmc.org/c/openbmc/bmcweb/+/84886
I'll also add it to 1120

baemyung replied:
The fix is included in this PR

baemyung committed:
Fix websocket error log handling

The commit b0ae71c[1] mishandles the error handling branches which
cause `operation_aborted` to be reported as an error.

```
            else if (ec != <error-cases>)
            {
                BMCWEB_LOG_ERROR("doRead error {}", ec);
            }
            else if (ec == boost::asio::error::operation_aborted)
            {
                BMCWEB_LOG_WARNING("doRead operation is aborted: {}", ec);
            }
```

This needs to handle `operation_aborted` first.

Tested:
- Unit test passes

[1] https://github.com/openbmc/bmcweb/commit/b0ae71c8f503b9e56a5bf88549346b15e7961e26

Change-Id: Ie4b0eafabcf41adf661ab3abc1fe1282c13d1d82
Signed-off-by: Myung Bae <myungbae@us.ibm.com>

baemyung commented:
Upstream:
- https://gerrit.openbmc.org/c/openbmc/bmcweb/+/84825
- https://gerrit.openbmc.org/c/openbmc/bmcweb/+/84819
- https://gerrit.openbmc.org/c/openbmc/bmcweb/+/84886

baemyung commented:
STG: https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=747613
